nbconvert/exporters/templateexporter.py

```diff
@@ -54,6 +56,29 @@
         'json_dumps': json.dumps,
 }
 
+class ExtensionTolerantLoader(BaseLoader):
+    """A template loader which optionally adds a given extension when searching.
+
+    Constructor takes two arguments: *loader* is another Jinja loader instance
+    to wrap. *extension* is the extension, which will be added to the template
+    name if finding the template without it fails. This should include the dot,
+    e.g. '.tpl'.
+    """
+    def __init__(self, loader, extension):
+        self.loader = loader
+        self.extension = extension
```

Comment:
I think if you change this to 

     def __init__(self, loader, extension_list):
          self.loader = loader
          self.extension_list = extension_list 

then you could accommodate (below) something that is similar to:

    if any([template.endswith(ext) for ext in self.extension_list]):
        raise
    return self.loader.get_source(environment, template+self.extension_list[0])

which would default to returning `.tpl`

and then later on you'd need to do something like:

    template_extension_list = List(['.tpl','tplx']).tag(config=True, affects_environment=True)